core/src/main/java/org/elasticsearch/cluster/routing/allocation/NodeRebalanceResult.java

```diff
@@ -73,16 +65,26 @@ public boolean isDeltaAboveThreshold() {
     }
 
     /**
-     * Returns the current weight of the node if the shard is not added to the node.
+     * Returns if the simulated weight of the shard after assigning the node to it would be better than the current node's weight.
+     * If {@code true}, then the shard should be rebalanced to the node, assuming allocation is allowed to the node.
      */
-    public float getCurrentWeight() {
-        return currentWeight;
+    public boolean isBetterWeightWithShardAdded() {
+        return betterWeightWithShardAdded;
     }
 
-    /**
-     * Returns the weight of the node if the shard is added to the node.
-     */
-    public float getWeightWithShardAdded() {
-        return weightWithShardAdded;
+    @Override
+    protected void innerToXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.field("delta_above_threshold", deltaAboveThreshold);
+        if (deltaAboveThreshold) {
+            builder.field("better_weight_with_shard_added", betterWeightWithShardAdded);
```

Comment:
Lets take two nodes, `A` and `B`, and we are trying to balance shard `S0` belonging to index `idx` which current resides on node `A`.  Lets say the current weight of node `A` with respect to `idx` is 25 and the current weight of node `B` with respect to `idx` is 10.  The current weight delta between the two nodes is 15.  Lets assume the configured threshold for prompting rebalances is 10, so node `B` will have `deltaAboveThreshold=true` with respect to shard `S0` on node `A`.  

Now, we simulate moving `S0` from `A` to `B`.  Lets say the new weight of `A` with respect to `idx` is now 10 and the new weight of `B` with respect to `idx` is 26.  The new delta between the two nodes with respect to `idx` is now 16 which is greater than the previous delta of 15, so `better_balance` would be false, even though `deltaAboveThreshold` is true.